Automattic/wp-calypso
# Editor: remove editor preference and related code

#### Changes proposed in this Pull Request

Removes the editor preference and related code, as it's no longer needed since Gutenberg became the default for all WP.com users ~1 year ago.

List of removed code:

- `getSelectedEditor` selector and tests, along with lingering uses of the selector that should no longer be relevant.
- `isClassicEditorForced` selector
- The `<QuerySiteSelectedEditor>` component and related `layout/query-selected-editor` feature flag used to trigger it.
- `requestSelectedEditor` and related plumbing for requesting the selected editor preference
- The classic block guide from wpcom-block-editor (see https://github.com/Automattic/wp-calypso/issues/46472 for context).
- Use of the `in-editor-deprecation-group` query param used to trigger the classic block guide.

Note that this PR incorporates the code from https://github.com/Automattic/wp-calypso/pull/55387

#### Testing instructions

* Test loading the editor from Calypso for, different kinds of sites (Simple, Atomic, Jetpack)
* Test that wpcom-block-editor loads without error for WP.com sites (see instructions at PCYsg-l4k-p2)
* Search wp-caypso code to ensure the removed selectors, action, component, etc are not used anywhere else

Fixes https://github.com/Automattic/wp-calypso/issues/46472
